recording-daemon/output.c

```diff
@@ -81,6 +81,19 @@ static void create_parent_dirs(char *dir) {
 	}
 }
 
+static output_t *output_alloc(const char *path, const char *name) {
+	output_t *ret = g_slice_alloc0(sizeof(*ret));
+	ret->file_path = g_strdup(path);
+	ret->file_name = g_strdup(name);
```

Comment:
In `recording-daemon/output.c`, function `void output_close(output_t *output)`, we have:
`g_clear_pointer(&output->file_name, g_free);`. shouldn't that be enough?

Reply:
No - if there wasn't a strdup there before and it is now, then there must be an extra free added somewhere. Looking closer at the code, I would say that the second argument to g_string_free (line 174/178) must be changed to TRUE so that the string buffer itself is also freed.

Reply:
I'm also confused a bit: if we have (line 88) `ret->full_filename = g_strdup_printf("%s/%s", path, name);`, shouldn't that also be `g_free`-ed? Instead we have `g_clear_pointer(&output->full_filename, g_free);` (line 337).

Reply:
g_clear_pointer(..., g_free) does that. But in your case, it's a strdup on a string that is already allocated (which is why there wasn't a strdup there before). If you change the second argument of g_string_free to TRUE then that does the g_free.

Reply:
OK, now I see your point, thanks.